components/scream/src/physics/p3/CMakeLists.txt

```diff
@@ -1,10 +1,60 @@
+set(CIMEROOT ${SCREAM_BASE_DIR}/../../cime)
+list(APPEND CMAKE_MODULE_PATH ${CIMEROOT}/src/CMake)
+
+set(GENF90 ${CIMEROOT}/src/externals/genf90/genf90.pl)
+set(ENABLE_GENF90 True)
+include(genf90_utils)
+include(Sourcelist_utils)
+
+# From cime/share/util/CMakeLists.txt
+set(share_util_path ${CIMEROOT}/src/share/util)
+set(genf90_files ${share_util_path}/shr_infnan_mod.F90.in ${share_util_path}/shr_assert_mod.F90.in)
+
+process_genf90_source_list("${genf90_files}" ${CMAKE_CURRENT_BINARY_DIR}
+  share_genf90_sources)
+
+sourcelist_to_parent(share_genf90_sources)
+
+list(APPEND share_sources "${share_genf90_sources}")
+
+list(APPEND share_sources
+  ${share_util_path}/shr_file_mod.F90
+  ${share_util_path}/shr_kind_mod.F90
+  ${share_util_path}/shr_const_mod.F90
+  ${share_util_path}/shr_sys_mod.F90
+  ${share_util_path}/shr_log_mod.F90
+  ${share_util_path}/shr_strconvert_mod.F90
+  ${share_util_path}/shr_nl_mod.F90
+  ${share_util_path}/shr_string_mod.F90
+  ${share_util_path}/shr_timer_mod.F90
+  ${share_util_path}/shr_abort_mod.F90
+  ${share_util_path}/shr_mpi_mod.F90
+  ${share_util_path}/shr_taskmap_mod.F90
+  ${share_util_path}/../timing/perf_mod.F90
+  ${share_util_path}/../timing/perf_utils.F90)
+
+# end share/util
+
+add_definitions(-DPCOLS=1 -DPSUBCOLS=1 -DPLEV=1)
```

Comment:
These definitions probably won't work. Not sure where I should get those values.